.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  test:
+    strategy:
+      matrix:
+        include:
+          - name: Tests
+            command: bin/tests --postgresql-host localhost
+          - name: Lint
+            command: make lint BINDIR="$(dirname $(which python))"
+          - name: Documentation
+            command: make docs BINDIR="$(dirname $(which python))"
+          - name: Dependencies
+            command: make github-actions-deps BINDIR="$(dirname $(which python))"
+          - name: Licenses
+            command: make licenses BINDIR="$(dirname $(which python))"
+          - name: Translations
+            command: make translations BINDIR="$(dirname $(which python))"
+          - name: Static Tests
+            command: bin/static_tests
+          - name: Static Pipeline
+            command: bin/static_pipeline
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:10.1
+        ports:
+          - 5432:5432
+    name: ${{ matrix.name }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Install platform dependencies
+        run: sudo apt -y install libcurl4-openssl-dev libssl-dev pkg-config
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8.2
+      - name: Cache Python dependencies
+        uses: actions/cache@v2
+        env:
+          cache-name: warehouse-cache-pip
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('requirements.txt', 'requirements/*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-${{ github.job }}-
+            ${{ runner.os }}-
+      - name: Install Python dependencies
+        run: |
+          pip install -U pip setuptools wheel
+          pip install -r requirements.txt
+          pip install -r requirements/dev.txt
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14.4.0
+      - name: Cache Node dependencies
+        uses: actions/cache@v2
+        env:
+          cache-name: warehouse-cache-npm
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ github.job }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install Node dependencies
+        run: npm ci
```

Comment:
Is this necessary in all the jobs? Should it be conditional?

Reply:
You're correct, but this wouldn't shorten the parallel runtime of the CI suite. I opted to keep this workflows simple instead.

Reply:
(because the longest-running job still requires the Node dependencies to be installed)

Reply:
I think it's worth saving the minutes because the number of parallel jobs affects other runs in the repo. This will be visible in other PRs when the jobs don't start for some time without any obvious reason. Besides, I'm pretty sure there's some monthly cap on the used minutes in the org...

Reply:
Thanks for pushing on this, this brought the CI run down to 3-ish minutes and also got me to make some progress towards #4948 as well.